src/lambdaisland/janus.clj

```diff
@@ -0,0 +1,87 @@
+(ns lambdaisland.janus
+  (:require [clojure.string :as str])
+  (:import [com.vladsch.flexmark.ast Heading BulletList BulletListItem]
+           [com.vladsch.flexmark.parser Parser]
+           [com.vladsch.flexmark.util.ast Block Node]))
+
+(set! *warn-on-reflection* true)
+
+(defn remove-nil-vals [item]
+  (into {} (filter (comp some? val)) item))
+
+(defn node-text
+  [^Block node]
+  (when node
```

Comment:
Check this here so `(node-text nil) ;;=> nil`, prevents us from having to do more nil checks elsewhere.

Also note the naming, functions like this should be named after what they return.